client/dashboard/profile-wizard/index.js

```diff
@@ -48,10 +62,51 @@ export default class ProfileWizard extends Component {
 		return currentStep;
 	}
 
+	goToNextStep() {
+		const currentStep = this.getCurrentStep();
+		const currentStepIndex = getSteps().findIndex( s => s.key === currentStep.key );
+		const nextStep = getSteps()[ currentStepIndex + 1 ];
+
+		if ( 'undefined' === nextStep ) {
+			this.updateProfile( { complete: true } );
+		}
+
+		return updateQueryString( { step: nextStep.key } );
+	}
+
+	updateProfile( params ) {
+		const { addNotice } = this.props;
+		const payload = stringifyQuery( params );
+
+		return apiFetch( {
+			path: `/wc-admin/v1/onboarding/profile${ payload }`,
```

Comment:
I doubt we would hit too long of a URL with the number of fields per page, but I'm still thinking we should update the controller to accept POST fields for the data instead of query strings.

Instead of `prepare_objects_query` in the REST controller, I think we could use `get_json_params`, which would allow us to pass the payload in the body.

Reply:
Sure!  Updated this to use `get_json_params`.  Still passed it through the existing `prepare_objects_query` so we can filter params.

I think this will also resolve #2299.

Reply:
Thanks for updating this! Confirmed this resolves #2299 by testing with this branch + Postman directly.